src/components/views/settings/tabs/user/EncryptionUserSettingsTab.tsx

```diff
@@ -33,19 +33,49 @@ import { ResetIdentityPanel } from "../../encryption/ResetIdentityPanel";
  *                        This happens when the user doesn't have a key a recovery key and the user clicks on "Set up recovery key" button of the RecoveryPanel.
  *  - "reset_identity": The panel to show when the user is resetting their identity.
  */
-type State = "loading" | "main" | "set_up_encryption" | "change_recovery_key" | "set_recovery_key" | "reset_identity";
+export type State =
+    | "loading"
+    | "main"
+    | "set_up_encryption"
+    | "change_recovery_key"
+    | "set_recovery_key"
+    | "reset_identity_compromised"
+    | "reset_identity_forgot";
 
-export function EncryptionUserSettingsTab(): JSX.Element {
-    const [state, setState] = useState<State>("loading");
-    const setUpEncryptionRequired = useSetUpEncryptionRequired(setState);
+interface Props {
+    /**
+     * If the tab should start in a state other than the deasult
+     */
+    initialState?: State;
+}
+
+export function EncryptionUserSettingsTab({ initialState = "loading" }: Props): JSX.Element {
+    const [state, setState] = useState<State>(initialState);
+    const matrixClient = useMatrixClientContext();
+
+    const recheckSetupRequired = useCallback(() => {
+        (async () => {
+            const crypto = matrixClient.getCrypto()!;
+            const isCrossSigningReady = await crypto.isCrossSigningReady();
+            if (isCrossSigningReady) {
+                setState("main");
+            } else {
+                setState("set_up_encryption");
+            }
+        })();
+    }, [matrixClient]);
```

Comment:
I'm in favour to use a custom hook like `useSetUpEncryptionRequired` before. It helps the readability of the component imo

Reply:
Something like this? I find the custom hook a bit awkward in this scenario since you have to pass a lot of functions around, but I don't mind too much.